src/h5web/providers/models.ts

```diff
@@ -50,7 +52,7 @@ export interface Link<T extends HDF5Link = HDF5Link> extends Entity {
 
 type PrimitiveType<T extends HDF5Type> = T extends HDF5NumericType
   ? number
-  : T extends HDF5StringType
+  : T extends HDF5StringType | HDF5ComplexType
```

Comment:
I would rather see a separate logical branch that returns `Complex`. Here you're saying that the value is a `string`, which is wrong. `ScalarVis` and `MatrixVis` should explicitely accept `Complex` and `Complex[]` values.